src/vss_tools/exporters/graphql.py

```diff
@@ -1,134 +1,532 @@
+        return text
+    words = re.sub(r"[^a-zA-Z0-9]", " ", text).split()
+    return "".join(word[0].upper() + word[1:] for word in words)
+
+
+def to_screaming_snake_case(text: str) -> str:
+    """Converts a string to screaming snake case (i.e., CAPITAL LETTERS)"""
+    text = re.sub(r"[^a-zA-Z0-9]", " ", text)
+    words = text.split()
+    return "_".join(word.upper() for word in words)
+
+
+def get_gql_name(text: str, gql_type: GQLElementType) -> str:
+    """
+    Converts a given text to a GraphQL compatible name based on the given GQLElementType.
+    For example:
+    - type, enum, and scalars must be PascalCase.
+    - fields and arguments must be camelCase.
+    - enum values must be SCREAMING_SNAKE_CASE.
+    """
+
+    if gql_type in {GQLElementType.FIELD, GQLElementType.ARGUMENT, GQLElementType.DIRECTIVE}:
+        text = to_camel_case(text)
+        return text if not keyword.iskeyword(text) else f"{text}_"
+    elif gql_type in {
+        GQLElementType.TYPE,
+        GQLElementType.ENUM,
+        GQLElementType.INTERFACE,
+        GQLElementType.UNION,
+        GQLElementType.SCALAR,
+    }:
+        text = text.replace(".", "_")
+        text = f"{text}_Enum" if gql_type == GQLElementType.ENUM else text
+        return text if not keyword.iskeyword(text) else f"{text}_"
+    elif gql_type in {GQLElementType.ENUM_VALUE}:
+        text = to_screaming_snake_case(text)
+        if text[0].isdigit():
+            text = f"_{text}"
+        return text if not keyword.iskeyword(text) else f"{text}_"
+    raise GraphQLExporterException(f"Invalid GQLElementType: {gql_type}")
+
+
+def get_unit_enum_name(quantity_kind: str) -> str:
+    """Get the name for the GraphQL unit enum"""
+    text = f"{quantity_kind}_Unit"
+    return get_gql_name(to_pascal_case(text), GQLElementType.ENUM)
+
+
+# ========= DataFrames for VSS metadata =========
+core_headers = ["fqn", "parent", "name", "type", "description", "comment", "deprecated"]
+leaf_specific_headers = ["datatype", "unit", "min", "max", "allowed", "default"]
+branch_specific_headers = ["instances"]
+
+
+def get_metadata_df(root: VSSNode) -> pd.DataFrame:
+    """Traverses the tree and returns a DataFrame with the metadata of all the nodes."""
+    headers = core_headers + leaf_specific_headers + branch_specific_headers
+
+    df = pd.DataFrame(columns=headers)
+
+    for node in PreOrderIter(root):
+        data = node.get_vss_data()
+        fqn = node.get_fqn()
+        parent = get_expected_parent(fqn)
+        name = node.name
+        vss_type = data.type.value
+        metadata = {}
+        for header in headers[2:]:
+            metadata[header] = getattr_nn(data, header, "")
+        metadata["fqn"] = fqn
+        metadata["parent"] = parent
+        metadata["name"] = name
+        metadata["type"] = vss_type
+        df = pd.concat([df, pd.DataFrame([metadata])], ignore_index=True)
+
+    # Check that the fqn is unique in the metadata
```

Comment:
Is there any reason to believe that this can happen? I do not think that is possible to have duplicated fqns. Did you find such a case?

Reply:
I haven't found that. But I am just making sure that it won't happen because I am relying on unique `fqn`

Reply:
Yeah then Just remove the check. It is named fqn for a reason. ;-)

Reply:
Because if we want to assure that then the right place would be in the general parsing of the model